### PR TITLE
76748 - erp facade to sap handshake testing

### PR DESCRIPTION
**Pull request by Vishal14583**
- Changes in xml template
- Changes in soap service as per new template.
- Few minor changes for Price information endpoint.
- Unit test cases changes

### CODE REVIEW

File: src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs [hunk 1]
```diff
@@ -28,27 +28,27 @@ public class ErpFacadeController : BaseController<ErpFacadeController>
...
+        public virtual async Task<IActionResult> PostPriceInformation([FromBody] JArray requestJson)
         {
-            var traceId = requestJson.SelectToken(TraceIdKey)?.Value<string>();
+            var corrId = requestJson.First.SelectToken(CorrIdKey)?.Value<string>();
 
-            if (string.IsNullOrEmpty(traceId))
+            if (string.IsNullOrEmpty(corrId))
             {
-                _logger.LogWarning(EventIds.TraceIdMissingInSAPEvent.ToEventId(), "TraceId is missing in the event received from the SAP.");
+                _logger.LogWarning(EventIds.TraceIdMissingInSAPEvent.ToEventId(), "CorrId is missing in the event received from the SAP.");
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs:37]: Can this be renamed as CorrIdMissinginSAPEvent

File: src/UKHO.SAP.MockAPIService/Models/z_adds_mat_info_Inteface.cs
Comment (Shirin14926) [inline src/UKHO.SAP.MockAPIService/Models/z_adds_mat_info_Inteface.cs:1]: Spell check the filename please- z_adds_mat_info_Interface.cs

File: src/UKHO.SAP.MockAPIService/Services/z_adds_mat_info.cs [hunk 2]
```diff
@@ -21,17 +22,22 @@ public z_adds_mat_info(IAzureBlobEventWriter azureBlobEventWriter, MockService m
...
 
-            Task.Run(async () => await _azureBlobEventWriter.UploadEvent(requestXML, z_ADDS_MAT_INFO.IM_MATINFO.CORRID, z_ADDS_MAT_INFO.IM_MATINFO.CORRID + '.' + RequestFormat));
+            Task.Run(async () => await _azureBlobEventWriter.UploadEvent(requestXML, iM_MATINFO.CORRID, iM_MATINFO.CORRID + '.' + RequestFormat));
 
             _mockService.CleanUp();
 
             return new Z_ADDS_MAT_INFOResponse()
             {
-                EX_MESSAGE = "Record successfully received for " + z_ADDS_MAT_INFO.IM_MATINFO.CORRID,
+                EX_MESSAGE = "Record successfully received for " + iM_MATINFO.CORRID,
```
Comment (Shirin14926) [inline src/UKHO.SAP.MockAPIService/Services/z_adds_mat_info.cs:40]: can we write message as "Record successfully received for CorrelationId :

File: tests/UKHO.ERPFacade.API.UnitTests/Controllers/ErpFacadeControllerTests.cs [hunk 3]
```diff
@@ -53,15 +53,15 @@ public async Task WhenValidRequestReceived_ThenErpFacadeReturns200OkResponse()
             A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
              && call.GetArgument<LogLevel>(0) == LogLevel.Information
              && call.GetArgument<EventId>(1) == EventIds.BlobExistsInAzure.ToEventId()
-             && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Blob exists in the Azure Storage for the trace ID received from SAP event.").MustHaveHappenedOnceExactly();
+             && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Blob exists in the Azure Storage for the corrId received from SAP event.").MustHaveHappenedOnceExactly();
         }
 
         [Test]
-        public async Task WhenTraceIdIsMissingInRequest_ThenErpFacadeReturnsReturns400BadRequestResponse()
+        public async Task WhenCorrIdIsMissingInRequest_ThenPostPriceInformationReturnsReturns400BadRequestResponse()
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.UnitTests/Controllers/ErpFacadeControllerTests.cs:60]: Can be renamed as WhenCorrIdIsMissingInRequest_ThenPostPriceInformationReturns400BadRequestResponse

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 4]
```diff
@@ -19,9 +17,9 @@ public class SAPXmlHelper
 
         static int actionCounter;
 
-        static List<string> AttrNotMatched = new List<string>();
+        static readonly List<string> AttrNotMatched = new List<string>();
         static List<string> ChangeAVCSUoS = new List<string>();
-        static Dictionary<string, List<string>> ChangeENCCell = new Dictionary<string, List<string>>();
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:24]: Can be simplified as 
static readonly Dictionary<string, List<string>> ChangeENCCell = new();

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 5]
```diff
@@ -19,9 +17,9 @@ public class SAPXmlHelper
...
+        static readonly Dictionary<string, List<string>> ChangeENCCell = new Dictionary<string, List<string>>();
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:22]: Similarly for other declarations in this file

**Review by Shirin14926 — COMMENTED** [review]
Few minor comments please

File: src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs (hunk 1 above)
Comment (Vishal14583) [reply src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs]: Updated

File: src/UKHO.SAP.MockAPIService/Models/z_adds_mat_info_Inteface.cs
Comment (Vishal14583) [reply src/UKHO.SAP.MockAPIService/Models/z_adds_mat_info_Inteface.cs:1]: Corrected

File: src/UKHO.SAP.MockAPIService/Services/z_adds_mat_info.cs (hunk 2 above)
Comment (Vishal14583) [reply src/UKHO.SAP.MockAPIService/Services/z_adds_mat_info.cs]: Updated

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs (hunk 4 above)
Comment (Vishal14583) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:24]: Updated

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs (hunk 5 above)
Comment (Vishal14583) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs]: Updated in all places in SAPXMLHelper

File: tests/UKHO.ERPFacade.API.UnitTests/Controllers/ErpFacadeControllerTests.cs (hunk 3 above)
Comment (Vishal14583) [reply tests/UKHO.ERPFacade.API.UnitTests/Controllers/ErpFacadeControllerTests.cs:60]: Renamed

**Review by Shirin14926 — APPROVED** [review]
LGTM. Thanks @Vishal14583 for fixing comments

**Review by jrippington — APPROVED** [review]
Looks fine to me!

**Review by HugoBurgess — APPROVED** [review]

**Pull request merged**